cli/flox/src/commands/activate.rs

```diff
@@ -297,47 +298,51 @@ impl Activate {
+            }
             if flox.features.services && !manifest.services.is_empty() && !in_place {
-                if self.start_services {
-                    supported_environment(&flox, &self.environment)?; // Error for remote envs.
-                }
-                tracing::debug!(
-                    start = self.start_services,
-                    socket_exists = socket_path.exists(),
-                    "setting service variables"
+                // Always set the socket var because we use it to register
+                // cleanup in tests
+                // We can probably drop this extra conditional once we drop
+                // `register_cleanup.sh`
+                exports.insert(
+                    FLOX_SERVICES_SOCKET_VAR,
+                    socket_path.to_string_lossy().to_string(),
                 );
-                if self.start_services && socket_path.exists() {
-                    debug!("detected existing services socket");
-                    debug!("will not start services");
-                    message::warning("Skipped starting services, services are already running");
-                } else {
-                    if self.start_services {
-                        debug!("will start services");
+                if self.start_services {
+                    let start_new_process_compose = if socket_path.exists() {
+                        shutdown_process_compose_if_all_processes_stopped(&socket_path)?
                     } else {
-                        debug!("will not start services");
-                    }
-                    exports.insert(
-                        FLOX_ACTIVATE_START_SERVICES_VAR,
-                        self.start_services.to_string(),
-                    );
-                    if !services_to_start.is_empty() {
+                        true
+                    };
+                    if start_new_process_compose {
+                        tracing::debug!(start = self.start_services, "setting service variables");
                         exports.insert(
-                            FLOX_SERVICES_TO_START_VAR,
-                            // Store JSON in an env var because bash doesn't
-                            // support storing arrays in env vars
-                            serde_json::to_string(&services_to_start)?,
+                            FLOX_ACTIVATE_START_SERVICES_VAR,
+                            self.start_services.to_string(), // "true"
                         );
+                        if !services_to_start.is_empty() {
+                            exports.insert(
+                                FLOX_SERVICES_TO_START_VAR,
+                                // Store JSON in an env var because bash doesn't
+                                // support storing arrays in env vars
+                                serde_json::to_string(&services_to_start)?,
+                            );
+                        }
```

Comment:
Nit: I wondered whether putting this in a conditional could cause us to leak the wrong values into a nested activation but upon trying it out (and finding https://github.com/flox/flox/issues/1937) I realised that we only set this value for ephemeral activations so it should never leak through to a "real" activation that is nested.

It might be worth guarding it with `&& is_ephemeral` and a comment to be explicit though.